### PR TITLE
enhancement(nginx_metrics source): comply with component spec

### PR DESCRIPTION
**Pull request by jdrouet**
- Update errors logs and events
- Update 
- Add `BytesReceived`
- Add `StreamClosedError`

Signed-off-by: Jérémie Drouet <jeremie.drouet@datadoghq.com>

### CODE REVIEW

File: src/internal_events/nginx_metrics.rs [hunk 1]
```diff
@@ -9,16 +9,31 @@ use crate::sources::nginx_metrics::parser::ParseError;
...
 pub struct NginxMetricsEventsReceived<'a> {
+    pub byte_size: usize,
     pub count: usize,
     pub uri: &'a str,
 }
 
 impl<'a> InternalEvent for NginxMetricsEventsReceived<'a> {
+    fn emit_logs(&self) {
+        trace!(
+            message = "Events received",
```
Comment (001wwang) [inline src/internal_events/nginx_metrics.rs:20]:
```suggestion
            message = "Events received.",
```
(for consistency)

File: src/internal_events/nginx_metrics.rs [hunk 2]
```diff
@@ -65,10 +94,24 @@ pub struct NginxMetricsStubStatusParseError<'a> {
...
+        );
     }
 
     fn emit_metrics(&self) {
+        counter!(
+            "component_errors_total", 1,
+            "endpoint" => self.endpoint.to_owned(),
+            "error" => self.error.to_string(),
+            "error_type" => "parse_failed",
+            "stage" => "processing",
```
Comment (001wwang) [inline src/internal_events/nginx_metrics.rs:112]: Should the `stage` here be `receiving` to match the associated log? (or vice versa)

**Review by 001wwang — APPROVED** [review]
LGTM, two minor edits recommended.

File: src/internal_events/nginx_metrics.rs (hunk 2 above)
Comment (jdrouet) [reply src/internal_events/nginx_metrics.rs:112]: you impress me to catch those 😅 thank you!

**Review by jszwedko — APPROVED** [review]
Looks good!

**Pull request merged**